### ISSUE TITLE
Syntax Error: Invalid Syntax

### ISSUE THREAD

**Issue by What1sBurnNotice** [shadowmoose/RedditDownloader#42]
I am on Mac OSX High Sierra and downloaded and unzipped the folder. I ran your command "_pip install -r requirements.txt_" after downloading Python. At first it didnt work so i ran "_sudo easy_install-3.7 pip_". it worked. But then when I tried running "_python main.py_" it gives me 
"_File "main.py", line 201
    print(("\t%3d:%-" + padding_len + "s -> ") % (i, name), end='')
                                                               ^
SyntaxError: invalid syntax_"

**Comment by shadowmoose** [comment]
Hey, can you first confirm your Python version by running ```python -V``` (note the uppercase 'V')? 

These issues are usually caused when you need to run the commands with ```python3``` instead, and the system defaults to an older version.

**Comment by What1sBurnNotice** [comment]
you are right when I run that command it shows Python 2.7.10. where should i go from here. 
I tried running _python3.7_ and it gave me 
_Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 26 2018, 23:26:24) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information._
Afterword I tried _python main.py_ which gave me 
_ File "<stdin>", line 1
    python main.py
              ^
SyntaxError: invalid syntax_

**Comment by shadowmoose** [comment]
Replace the ```python``` in the ```python main.py``` command with the correct version installed on your machine for python 3+.

Usually, this would be called by running ```python3 main.py``` instead.

**Comment by What1sBurnNotice** [comment]
I ran both `python3 main.py` and `python3.7 main.py` the results are the same as follows:
```
_(My)-MacBook-Air:RedditDownloader-master (Me)$ python3.7 main.py
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144, got 128
  return f(*args, **kwds)

====================================
    Reddit Media Downloader 2.2
====================================
    (By ShadowMoose @ Github)

Failed to load settings file! A new one will be generated!
Would you like to start the WebUI to help set things up?([y]/n):_ 
```

```
How would you like to open the UI?
	1: Chrome Application Mode (recommended)
	2: The default system browser
	3: Don't auto-open a browser
```

So i input "1", "2", and "3" on separate run throughs to see if any worked each time was the same:

```
-Saved Settings-
BOOTSTRAPPING RMD...
Launching:  ('/Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7', ['/Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7', 'main.py', '--no_restart'])
Traceback (most recent call last):
  File "main.py", line 146, in <module>
    ret = subprocess.call(sargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gevent/subprocess.py", line 251, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gevent/subprocess.py", line 463, in __init__
    hub = get_hub()
  File "src/gevent/_hub_local.py", line 73, in gevent.__hub_local.get_hub_noargs
  File "src/gevent/_hub_local.py", line 80, in gevent.__hub_local.get_hub_noargs
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gevent/hub.py", line 390, in __init__
    WaitOperationsGreenlet.__init__(self, None, None)
  File "src/gevent/_greenlet_primitives.py", line 42, in gevent.__greenlet_primitives.TrackedRawGreenlet.__init__
SystemError: /Users/sysadmin/build/v3.7.0/Objects/dictobject.c:1439: bad argument to internal function
```

Side note: sorry to be a bother as you can tell im not great with Python or coding in general

**Comment by What1sBurnNotice** [comment]
Help please

**Comment by What1sBurnNotice** [comment]
hello? anyone there?

**Comment by shadowmoose** [comment]
Hi, I haven't been around my setup for a couple weeks now, but I'll try to resolve this anyways. This is a bug in one of the libraries RMD uses. A quick search shows it is a known library bug [here](https://github.com/gevent/gevent/issues/1260). This seems to be fixed, but it doesn't look like they've released the fix in a new version yet, so my best suggestion is to try updating the dependencies on your install (using pip). 

If this doesn't resolve the issue, it will likely be fixed soon. I'll also be rolling out alternate ways to launch RMD in the near future (when I can find some free time), which will fix the issue in a roundabout way.

**Comment by shadowmoose** [comment]
Resolved as of [Release 3.0.0](https://github.com/shadowmoose/RedditDownloader/releases/tag/3.0.0). Feel free to reopen if this still needs more attention.
